Mapper/BaseGroupedMapper.php

```diff
@@ -70,11 +70,17 @@ public function with($name, array $options = array())
             'collapsed' => false,
             'class' => false,
             'description' => false,
+            'label' => $this->admin->getLabelTranslatorStrategy()->getLabel($name, $this->getName(), 'group'),
             'translation_domain' => null,
             'name' => $name,
             'box_class' => 'box box-primary',
         );
 
+        // BC layer for manual label
+        if (strpos($name, '.')) {
+            $defaultOptions['label'] = $name;
+        }
```

Comment:
I think you should go on with your idea of a parameter, and remove this, the config parameter is the only clean solution I can imagine.

So: 
- in 3.x `translate_group_label`: defaults to false, and raises a deprecation warning in that case.
- In 4.x : completely ignored, raises a deprecation warning when set
- In 5.x: removed